packages/context-editor/src/components/MenuBar.tsx

```diff
@@ -128,35 +153,67 @@ const ParagraphDropdown = () => {
 	);
 };
 
-export const MenuBar = () => {
+const MenuItemButton = ({ menuItem }: { menuItem: MenuItem }) => {
 	const { view } = usePluginViewContext();
+	const { key, name, icon, command } = menuItem;
+	const { run, canRun, isActive } = command(view)(view.state);
+	return (
+		<Button
+			key={key}
+			onClick={() => {
+				view.focus();
+				run();
+			}}
+			variant="ghost"
+			size="sm"
+			disabled={!canRun}
+			type="button"
+			className={cn("h-fit rounded-sm p-1", {
+				"bg-blue-200 hover:bg-blue-300": isActive,
+			})}
+			title={name}
+		>
+			{icon}
+		</Button>
+	);
+};
+
+const Separator = () => {
+	return (
+		<div className="flex items-center px-4">
+			<div className="h-6 w-px bg-gray-300" />
+		</div>
+	);
+};
+
+export const MenuBar = () => {
 	return (
 		<div
-			className="flex items-center rounded border bg-slate-50"
+			className="flex items-center rounded-t border bg-slate-50 p-4"
```

Comment:
Really great job with all the `MenuBar` changes, looks so nice now!

tiny tiny nitpick: the color in the designs is `gray-50` instead of `slate-50`

```suggestion
			className="flex items-center rounded-t border bg-gray-50 p-4"
```

(got lost in the review bc i commented on the wrong line)